packages/notifi-dapp-example/src/components/NotifiSignUpButton.tsx

```diff
@@ -0,0 +1,162 @@
+import { useGlobalStateContext } from '@/context/GlobalStateContext';
+import { formatTelegramForSubscription } from '@/utils/stringUtils';
+import {
+  CardConfigItemV1,
+  EventTypeConfig,
+} from '@notifi-network/notifi-frontend-client';
+import {
+  createConfigurations,
+  subscribeAlertsByFrontendClient,
+  useFrontendClientLogin,
+  useNotifiClientContext,
+  useNotifiForm,
+  useNotifiSubscribe,
+  useNotifiSubscriptionContext,
+} from '@notifi-network/notifi-react-card';
+import { isValidPhoneNumber } from 'libphonenumber-js';
+import { useRouter } from 'next/navigation';
+import React, { useCallback, useMemo } from 'react';
+
+export type NotifiSignUpButtonProps = Readonly<{
+  buttonText: string;
+  data: CardConfigItemV1;
+}>;
+
+type TargetGroupData = {
+  name: string;
+  emailAddress?: string;
+  phoneNumber?: string;
+  telegramId?: string;
+  discordId?: string;
+};
+
+//todo implement in card context
+const inputs = {};
+
+export const NotifiSignUpButton: React.FC<NotifiSignUpButtonProps> = ({
+  buttonText,
+  data,
+}) => {
+  const eventTypes = data.eventTypes;
+  const { isInitialized, subscribe, updateTargetGroups } = useNotifiSubscribe({
+    targetGroupName: 'Default',
+  });
+  const frontendClientLogin = useFrontendClientLogin();
+
+  const router = useRouter();
+
+  const { frontendClient } = useNotifiClientContext();
+
+  const {
+    connectedWallets,
+    loading,
+    setCardView,
+    useDiscord,
+    render,
+    setLoading,
+    syncFtuStage,
+  } = useNotifiSubscriptionContext();
+
+  const { setIsGlobalLoading } = useGlobalStateContext();
+
+  const { formErrorMessages, formState } = useNotifiForm();
+
+  const { phoneNumber, telegram: telegramId, email } = formState;
+
+  const { email: emailErrorMessage, phoneNumber: smsErrorMessage } =
+    formErrorMessages;
+
+  const targetGroup: TargetGroupData = useMemo(
+    () => ({
+      name: 'Default',
+      emailAddress: email === '' ? undefined : email,
+      phoneNumber: isValidPhoneNumber(phoneNumber) ? phoneNumber : undefined,
+      telegramId:
+        telegramId === ''
+          ? undefined
+          : formatTelegramForSubscription(telegramId),
+      discordId: useDiscord ? 'Default' : undefined,
+    }),
+    [email, phoneNumber, telegramId, useDiscord],
+  );
+
+  const renewTargetGroups = useCallback(async () => {
+    return updateTargetGroups();
+  }, [updateTargetGroups]);
+
+  const subscribeAlerts = useCallback(
+    async (eventTypes: EventTypeConfig, inputs: Record<string, unknown>) => {
+      return subscribe(
+        createConfigurations(eventTypes, inputs, connectedWallets),
+      );
```

Comment:
Instead of using hooks (Which is deprecated), we shall make use of `frontendClient` object.